ck.sh

```diff
@@ -0,0 +1,216 @@
+#!/bin/sh
+
+set -e
+
+function init() {
+	# Make sure submodule is registered and up-to-date.
+	git submodule update --init
+
+	# Fetch remote changes.
+	cd ckeditor-dev
+	# not needed git fetch
+
+	# Make sure our working copy is clean.
+	git reset --hard HEAD --quiet
+	git clean -fdx
+}
+
+function message() {
+	if [[ ! -z "$1" ]] ; then
+		echo
+		printf -- '-%.0s' {1..80}; echo ""
+		printf "%*s\n" $(( ( $(echo $1 | wc -c ) + 80 ) / 2 )) "$1"
+		printf -- '-%.0s' {1..80}; echo ""
+		echo
+
+		if [[ ! -z  "$2" ]] ; then
+			echo
+			echo "$2" | fold -w 80
+			echo
+		fi
+	fi
+}
+
+function usage() {
+	echo "Usage: ck.sh COMMAND"
+	echo
+	echo "  Where COMMAND is either:"
+	echo
+	echo "  🔧  setup: Setup everything to start working on a patch"
+	echo "  💉  patch: Generate patches "
+	echo "  🔥  build: Generate a patched version of CKEditor"
+	echo
+	echo
+}
+
+# Check arguments
+if [ $# -ne 1 ]; then
+	usage
+  exit 1
+fi
+
+COMMAND=$1
+
+case "$COMMAND" in
+	build)
+		VERSION=$2
+
+		init
+		git checkout --detach HEAD --quiet
+
+		# Warn the user and prompt to continue
+		message "⚠️  WARNING"
+
+		str=$(printf "%s\n" "This will generate a patched version of CKEditor" \
+			"Are you sure you want to continue?")
+
+		read -p "$str [y/n]? " yn
+
+		case $yn in
+			[Yy]*)
+				git branch -f liferay HEAD
+				git checkout liferay
+
+				echo
+				echo "Checking for existing patches"
+				echo
+
+				if ! ls ../patches/*; then
+					echo "There doesn't seem to be any patch"
+					# TODO: Probably exit?
+					echo
+				fi
+
+				echo
+				echo "Applying patches from \"patches/\" directory."
+				echo
+
+				if ! git am ../patches/*; then
+					echo
+					echo "⚠️ There was a problem applying patches:"
+					echo
+					echo "To retry manually and fix:"
+					echo
+					echo "  cd ckeditor-dev"
+					echo "  git am --abort"
+					echo "  git am ../patches/*"
+					echo
+					echo "Once you are happy with the result, run \`sh ./ck.sh patch\` to update the contents of \"patches/\"."
+					echo
+					exit 1
+				fi
+
+				if [ -n "$DEBUG" ]; then
+					dev/builder/build.sh --build-config ../../../build-config.js \
+						--leave-css-unminified --leave-js-unminified
+				else
+					dev/builder/build.sh --build-config ../../../build-config.js
+				fi
+
+				# Remove old build files.
+				rm -rf ../ckeditor/*
+
+				# Replace with new build files.
+				cp -r dev/builder/release/ckeditor/* ../ckeditor/
+
+				str=$(printf "%s\n" "Don't forget to commit the result!\n\n" \
+					"    git add -A -- ckeditor\n" \
+					"    git commit -m 'Update CKEDITOR'")
+
+				message "✅  DONE" "$str"
+				;;
+			*)
+				echo
+				echo "Aborting."
+				echo
+		esac
+		;;
+	patch)
+		init
+		# Navigate back to superproject
+		cd ..
```

Comment:
It might be nice to make `init` do that on its own. It is always a bit surprising when calling a function leaves you in a different working directory.